Add DRS compact identifier support

dannon opened the pull request:
Support compact identifiers (e.g., drs://drs.anv0:object-id) by resolving prefixes via identifiers.org. Includes caching and tests.

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

dannon committed:
Two minor typos in drs utils

dannon committed:
Add DRS compact identifier support

Support compact identifiers (e.g., drs://drs.anv0:object-id) by
resolving prefixes via identifiers.org. Includes caching and tests.

jmchilton commented on package.json:
```diff
@@ -19,5 +19,6 @@
     "cpy-cli": "^5.0.0"
-  }
+  },
+  "packageManager": "yarn@1.22.22+sha512.a6b2f7906b721bba3d67d4aff083df04dad64c399707841b7acf00f6b133b7ac24255f2652fa22ae3534329dc6180534e98d17432037ff6fd140556e2bb3137e"
```
Probably unrelated.

dannon replied:
Yep, oops.  I'll pop it out

jmchilton reviewed (approved):
Very exciting - thanks for the tests!

dannon committed:
Improve DRS util code quality

Refactor S3 download logic, add type hints, enhance error handling

dannon committed:
Formatting and linting post-rebase (dep upgrade tweaked it?)

dannon committed:
Fix DRS tests to mock access endpoint

The access_id resolution logic now prefers fetching from /access/{id}
when present, so tests need to mock that endpoint rather than providing
inline access_url in the object response.

dannon committed:
Fix access_id/access_url check for None values

Galaxy's DRS response can include access_id: None, so we need to check
for truthy values, not just key existence.
